Controller/Cart/Data.php

```diff
@@ -119,8 +119,10 @@ public function execute()
                 'cartReference' => $cartReference,
             ];
 
-            // cart data validation requirement
-            $cart['currency']->currency_code = @$cart['currency']->currency;
+            if (@$cart['currency']->currency) {
+                // cart data validation requirement
+                $cart['currency']->currency_code = @$cart['currency']->currency;
```

Comment:
@ is not needed at this point in front of the $cart if it passes the if condition

Reply:
Updated.